src/doc/trpl/arrays-vectors-and-slices.md

```diff
@@ -60,6 +59,16 @@ let v = vec![1, 2, 3]; // v: Vec<i32>
 brackets `[]` with `vec!`. Rust allows you to use either in either situation,
 this is just convention.)
 
+Vectors are to arrays what `String` is to `&str`. While the arrays are
```

Comment:
This isn't correct. There's three core array-ish types:

* [T; n]: Array
* [T]: Slice
* Vec<T>: Vector

[T; n] is "on the stack", because it has a compile-time-constant size. 

[T] (usually seen as &[T]), is a dynamically-sized type because it is a handle to some compile-time-unknown number of elements. In that sense, [T] can't ever exist on its own; it must exist behind a pointer as e.g. Box<[T]> or &[T].  Any such pointer becomes "fat" as it includes the number of elements in the slice as run-time information, as well as the actual address where the slice starts. A slice may just as easily refer to the contents of a Vec or Array.

Vec is a heap-allocated growable array, conceptually similar to a Box<[T]>, but without the requirement that capacity = length.

`Vec == String`, and `&str  == &[T]`, not `[T; n]`. There is no string analog to `[T; n]`.